flocker/ca/test/test_ca.py

```diff
@@ -21,6 +23,297 @@
+        """
+        A ``PathError`` is raised if the key file path given to
+        ``NodeCredential.from_path`` does not exist.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        # random UUID that doesn't exist in path
+        uuid = self.uuid
+        cert_file = b"{uuid}.crt".format(uuid=uuid)
+        key_file = b"{uuid}.key".format(uuid=uuid)
+        crt_path = path.child(cert_file)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        e = self.assertRaises(
+            PathError, NodeCredential.from_path, path, uuid
+        )
+        expected = (b"Private key file could not be opened. "
+                    b"No such file or directory "
+                    b"{path}").format(
+                        path=path.child(key_file).path)
+        self.assertEqual(str(e), expected)
+
+    @not_root
+    @skip_on_broken_permissions
+    def test_load_error_on_unreadable_certificate_file(self):
+        """
+        A ``PathError`` is raised if the certificate file path given to
+        ``NodeCredential.from_path`` cannot be opened for reading.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        # random UUID that doesn't exist in path
+        uuid = self.uuid
+        cert_file = b"{uuid}.crt".format(uuid=uuid)
+        key_file = b"{uuid}.key".format(uuid=uuid)
+        crt_path = path.child(cert_file)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        # make file unreadable
+        crt_path.chmod(64)
+        key_path = path.child(key_file)
+        key_file = key_path.open(b'w')
+        key_file.write(b"dummy")
+        key_file.close()
+        # make file unreadable
+        key_path.chmod(64)
+        e = self.assertRaises(
+            PathError, NodeCredential.from_path, path, uuid
+        )
+        expected = (
+            b"Certificate file could not be opened. "
+            b"Permission denied {path}"
+        ).format(path=crt_path.path)
+        self.assertEqual(str(e), expected)
+
+    @not_root
+    @skip_on_broken_permissions
+    def test_load_error_on_unreadable_key_file(self):
+        """
+        A ``PathError`` is raised if the key file path given to
+        ``NodeCredential.from_path`` cannot be opened for reading.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        # random UUID that doesn't exist in path
+        uuid = self.uuid
+        cert_file = b"{uuid}.crt".format(uuid=uuid)
+        key_file = b"{uuid}.key".format(uuid=uuid)
+        crt_path = path.child(cert_file)
+        crt_file = crt_path.open(b'w')
+        crt_file.write(b"dummy")
+        crt_file.close()
+        key_path = path.child(key_file)
+        key_file = key_path.open(b'w')
+        key_file.write(b"dummy")
+        key_file.close()
+        # make file unreadable
+        key_path.chmod(64)
```

Comment:
More usual to express this in octal (0100). Is there a reason to keep user-executable bit set?